lib/broccoli/bundler.js

```diff
@@ -149,7 +152,24 @@ module.exports = class Bundler {
 
     let scriptOutputFiles = options.scriptOutputFiles;
 
-    let appJs = concat(applicationAndDepsTree, {
+    let name = this.name;
+    let tree = applicationAndDepsTree;
+    if (experiments.TREE_SHAKING) {
+      tree = new BroccoliEvict(applicationAndDepsTree, {
```

Comment:
per conversation with @rwjblue, we should likely create `broccoli-assembler` & move `new Assembler(...)` & `concat` code there.

Reply:
make sure to mention timings.